src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -57,93 +62,204 @@ cdef class _AioCall:
+                _UNKNOWN_CANCELLATION_DETAILS,
+                None,
+                None,
+            )
 
-        cdef tuple operations
-        cdef Operation initial_metadata_operation
-        cdef Operation send_message_operation
-        cdef Operation send_close_from_client_operation
-        cdef Operation receive_initial_metadata_operation
-        cdef Operation receive_message_operation
-        cdef Operation receive_status_on_client_operation
+    async def unary_unary(self,
+                          bytes method,
+                          bytes request,
+                          object deadline,
+                          object cancellation_future,
+                          object initial_metadata_observer,
+                          object status_observer):
+        """Performs a unary unary RPC.
+        
+        Args:
+          method: name of the calling method in bytes.
+          request: the serialized requests in bytes.
+          deadline: optional deadline of the RPC in float.
+          cancellation_future: the future that meant to transport the
+            cancellation reason from the application layer.
+          initial_metadata_observer: a callback for received initial metadata.
+          status_observer: a callback for received final status.
+        """
+        cdef tuple ops
 
-        cdef char *c_details = NULL
+        cdef SendInitialMetadataOperation initial_metadata_op = SendInitialMetadataOperation(
+            _EMPTY_METADATA,
+            GRPC_INITIAL_METADATA_USED_MASK)
+        cdef SendMessageOperation send_message_op = SendMessageOperation(request, _EMPTY_FLAGS)
+        cdef SendCloseFromClientOperation send_close_op = SendCloseFromClientOperation(_EMPTY_FLAGS)
+        cdef ReceiveInitialMetadataOperation receive_initial_metadata_op = ReceiveInitialMetadataOperation(_EMPTY_FLAGS)
+        cdef ReceiveMessageOperation receive_message_op = ReceiveMessageOperation(_EMPTY_FLAGS)
+        cdef ReceiveStatusOnClientOperation receive_status_on_client_op = ReceiveStatusOnClientOperation(_EMPTY_FLAGS)
 
-        initial_metadata_operation = SendInitialMetadataOperation(_EMPTY_METADATA, GRPC_INITIAL_METADATA_USED_MASK)
-        initial_metadata_operation.c()
+        ops = (initial_metadata_op, send_message_op, send_close_op,
+               receive_initial_metadata_op, receive_message_op,
+               receive_status_on_client_op)
 
-        send_message_operation = SendMessageOperation(request, _EMPTY_FLAGS)
-        send_message_operation.c()
+        try:
+            self._create_grpc_call(deadline, method)
+            try:
+                await callback_start_batch(self._grpc_call_wrapper,
+                                           ops,
+                                           self._loop)
+            except asyncio.CancelledError:
+                status = self._cancel_and_create_status(cancellation_future)
+                status_observer(status)
+                raise
```

Comment:
I'm still skeptic on this pattern, it means that callers and interceptors would need to do so for observing the code:

```python
call = hi(...)
try:
   await call
except grpc.RpcError as err:
    status = err.code()
    raise
except CancelledError as cancellation:
    status = await call.code()
    raise
else:
    status = await call.done()
```
So the way of getting the canceled error is by checking the status of the call since the exception is not reported as a generic `RpcError`.  And with the implementation of interceptors, because a cancellation can happen after and before the real RPC call, the code returned in the context of a `CancelledError` could be different than the `rpc.StatusCode.CANCELLED`

As I said, I'm skeptic but I do not have any other good solution :(. And with this, we provide the way of not masking the `CanclledError` and giving the chance to the user for gathering the status code.

Reply:
I do not like either :)

if I would have to choose I would prefer to not touch the `CancelledError` and make explicit that in case of a cancellation the code would need to be gathered from the `Call` object.

But just my personal taste.

Reply:
Do you mean my current solution is okay to you?

Reply:
yes!